pallets/author-mapping/src/lib.rs

```diff
@@ -56,15 +60,46 @@ pub mod pallet {
 		/// Currency in which the security deposit will be taken.
 		type DepositCurrency: Currency<Self::AccountId> + ReservableCurrency<Self::AccountId>;
 		/// The amount that should be taken as a security deposit when registering an AuthorId.
-		type DepositAmount: Get<<Self::DepositCurrency as Currency<Self::AccountId>>::Balance>;
+		type DepositAmount: Get<BalanceOf<Self>>;
 
 		/// A rough preliminary check to determine whether an account can make a new registration.
 		/// If you don't wish to do any such check, just return `true`.
 		fn can_register(account: &Self::AccountId) -> bool;
 	}
 
 	#[pallet::hooks]
-	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_runtime_upgrade() -> Weight {
+			//TODO should we consier versioning here?
+			// For example make a storage item to track the pallet version and only upgrade if necessary?
```

Comment:
I think this is necessary? Otherwise it means that we have to remember to remove this part after the upgrade right ?
What does parity usually do ?

Reply:
Yeah, I agree it means we have to remove this part. Apparently for Polkadot they remember to remove them afterwards.

![image](https://user-images.githubusercontent.com/2915325/120342864-2ed5aa80-c2c6-11eb-85f7-2c996204cf83.png)